api/v1beta3/ratelimitpolicy_types.go

```diff
@@ -46,6 +41,9 @@ const (
 var (
 	RateLimitPolicyGroupKind  = schema.GroupKind{Group: SchemeGroupVersion.Group, Kind: "RateLimitPolicy"}
 	RateLimitPoliciesResource = SchemeGroupVersion.WithResource("ratelimitpolicies")
+	// Top level predicate rules key starting with # to prevent conflict with limit names
+	// TODO(eastizle): this coupling between limit names and rule IDs is a bad smell. Merging implementation should be enhanced.
```

Comment:
We can ID the rules however we want, as long as each rule has a unique ID. That was the case of limits as the only kind of rule before this PR. It made sense IDing the rule after the unique limit name rather than coming up with something different and possibly more opaque for the rule ID.

Now, we want to differentiate these two kinds of rules (conditions and limits) within a RateLimitPolicy spec. Similarly to how we handle this in the AuthPolicy (where there are 10 different kinds of rules), we can define different prefixes for each kind. E.g. conditions could be `conditions#<condition-index>` and limits `limits#<limit-name>`.

Sure, the new rule ID limits prefix would ultimately show up in the Limitador CR and wasm config as `limit.limits_<limit-name>__<hash:rlp_ns/rlp_name>`. To avoida that, a trick that I think would work is setting the prefix in `Rules()`/`SetRules()` to `limit.` and changing [this line](https://github.com/Kuadrant/kuadrant-operator/blob/3b8e313d552090c52d8aadca95f6952f42a03192/controllers/ratelimit_workflow_helpers.go#L69) to `identifier := ""`.